### PR TITLE
MONGOID-5323: Remove deprecated Mongoid::Errors::EagerLoad class

### PR DESCRIPTION
**Pull request by johnnyshields**
It is not used in the code at all.

Don't think this needs to be added to release notes (as no one is likely to be using it), but happy to add it if you'd like.

### CODE REVIEW

File: lib/mongoid/errors/eager_load.rb [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-      Mongoid.deprecate(self, :initialize)
-
-      # Create the new eager load error.
-      #
-      # @example Create the new eager load error.
-      #   EagerLoad.new(:preferences)
-      #
-      # @param [ Symbol ] name The name of the association.
-      def initialize(name)
-        super(compose_message("eager_load", { name: name }))
```
Comment (p-mongo) [inline lib/mongoid/errors/eager_load.rb:21]: This message needs to be removed from `lib/config/locales/en.yml` also.

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-5323

File: lib/mongoid/errors/eager_load.rb (hunk 1 above)
Comment (johnnyshields) [reply lib/mongoid/errors/eager_load.rb:21]: Good catch. Fixed.

**Comment by p-mongo** [comment]
The require needs to be removed from `errors.rb`.

**Comment by johnnyshields** [comment]
> The require needs to be removed from `errors.rb`.

Done

**Review by Neilshweky — APPROVED** [review]
This looks good to me, I am re-running the evergreen tests to make sure they work correctly.

**Comment by Neilshweky** [comment]
@johnnyshields this one needs to be rebased as well

**Comment by johnnyshields** [comment]
Rebase done.

**Review by Neilshweky — APPROVED** [review]
@p-mongo release notes look good to me

**Pull request merged**